src/Workspaces/SharedUtilitiesAndExtensions/Workspace/VisualBasic/Extensions/CastExpressionSyntaxExtensions.vb

```diff
@@ -10,21 +10,127 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions
         Public Function Uncast(cast As PredefinedCastExpressionSyntax) As ExpressionSyntax
-            Return Uncast(cast, cast.Expression)
+            Return Uncast(cast, cast.OpenParenToken, cast.Expression, Nothing, Nothing, cast.CloseParenToken)
         End Function
 
-        Private Function Uncast(castNode As ExpressionSyntax, innerNode As ExpressionSyntax) As ExpressionSyntax
-            Dim resultNode = innerNode.WithTriviaFrom(castNode)
+        Private Function Uncast(castNode As ExpressionSyntax, openParen As SyntaxToken, innerNode As ExpressionSyntax, commaToken As SyntaxToken, typeNode As TypeSyntax, closeParen As SyntaxToken) As ExpressionSyntax
+
+            Dim leadingTrivia As List(Of SyntaxTrivia) = castNode.GetLeadingTrivia.ToList
+            If openParen.LeadingTrivia.ContainsCommentOrLineContinue Then
+                leadingTrivia.AddRange(openParen.LeadingTrivia)
+            End If
+
+            If openParen.TrailingTrivia.ContainsCommentOrLineContinue Then
+                leadingTrivia.AddRange(openParen.TrailingTrivia)
+            End If
+            If (Not castNode.GetLeadingTrivia.Any) AndAlso leadingTrivia.FirstOrDefault.IsKind(SyntaxKind.WhitespaceTrivia) Then
+                leadingTrivia.RemoveAt(0)
+            End If
+
+            If innerNode.GetLeadingTrivia.ContainsCommentOrLineContinue Then
+                leadingTrivia.AddRange(innerNode.GetLeadingTrivia)
+            End If
+
+            Dim trailingTrivia As List(Of SyntaxTrivia) = castNode.GetTrailingTrivia.ToList
+            If closeParen.TrailingTrivia.ContainsCommentOrLineContinue Then
+                trailingTrivia.InsertRange(0, closeParen.TrailingTrivia)
+            End If
+
+            If closeParen.LeadingTrivia.ContainsCommentOrLineContinue Then
+                trailingTrivia.InsertRange(0, openParen.LeadingTrivia)
+            End If
+
+            If typeNode IsNot Nothing Then
+                If typeNode.GetTrailingTrivia.ContainsCommentOrLineContinue Then
+                    trailingTrivia.InsertRange(0, typeNode.GetTrailingTrivia)
+                End If
+
+                If typeNode.GetLeadingTrivia.ContainsCommentOrLineContinue Then
+                    trailingTrivia.InsertRange(0, typeNode.GetLeadingTrivia)
+                End If
+            End If
+
+            If commaToken.IsKind(SyntaxKind.CommaToken) Then
+                If commaToken.TrailingTrivia.ContainsCommentOrLineContinue Then
+                    trailingTrivia.InsertRange(0, commaToken.TrailingTrivia)
+                End If
+
+                If commaToken.LeadingTrivia.ContainsCommentOrLineContinue Then
+                    trailingTrivia.InsertRange(0, commaToken.LeadingTrivia)
+                End If
+            End If
+
+            If innerNode.GetTrailingTrivia.ContainsCommentOrLineContinue Then
+                trailingTrivia.InsertRange(0, innerNode.GetTrailingTrivia)
+            End If
+
+            If trailingTrivia.Count > 0 Then
```

Comment:
Todd is far too complicated.

Reply:
It needs to handle all the ways comments and _ can appear and VB is not a flexible as C# as to where comments can appear  and I have tests that hit each case.

Reply:
needs a lot of coments on its own then stating what i going on, with exmaples of the before and what the after is.